src/NuGet.Core/NuGet.Commands/Services/PackageMetadataService.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Commands.Experimental.Services
+{
+    public class PackageMetadataService : IPackageMetadataService
+    {
+        private readonly ILogger _logger;
+        private readonly IEnumerable<Lazy<INuGetResourceProvider>> _resourceProviders;
+
+        public PackageMetadataService(IEnumerable<Lazy<INuGetResourceProvider>> resourceProviders, ILogger logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _resourceProviders = resourceProviders ?? throw new ArgumentNullException(nameof(resourceProviders));
+        }
+
+        public PackageMetadataService(ILogger logger)
```

Comment:
My instinct is that the logger is probably more relevant per operation instead of more globally, but I don't imagine this decision was taken lightly, but something to watch for.